### PR TITLE
Initial functionality for source panel

### PR DESCRIPTION
**Pull request by conor-jh**
This PR should close out #2 and #24 

## Overview of changes
- Refactor parts of `RequestBar` to handle setting data in panel state
- Give each data tree a unique ID based on total tree count
- Renders `RequestBar` and `DataCanvas` to source panel
- Displays empty state when no data exists

### CODE REVIEW

File: src/main/containers/TestPanel.jsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ const TestPanel = (props) => {
   return (
     <section className='panel' >
       <p>Test panel</p>
-      <ViewTest testdata={largeData}/>
+      {/* <DataTree data={largeData}/> */}
```
Comment (Donaldblanc) [inline src/main/containers/TestPanel.jsx:14]: Did we comment this out for testing ? or are we no longer going to render the tree here ?

**Review by Donaldblanc — APPROVED** [review]
Looks good, the project is coming along well. I am interested in seeing this scale as the user appends additional tests.

File: src/main/containers/SourcePanel.jsx [hunk 2]
```diff
@@ -1,18 +1,19 @@
 import React, { useState } from 'react';
+import DataCanvas from './DataCanvas';
```
Comment (Donaldblanc) [inline src/main/containers/SourcePanel.jsx]: for consistency and the linter, we should append the .jsx file type when importing.

**Review by Donaldblanc — CHANGES_REQUESTED** [review]
after the minor file type change, this PR should be good to go

**Review by joepav — APPROVED** [review]
Look good to me for now!

File: src/main/containers/TestPanel.jsx (hunk 1 above)
Comment (conor-jh) [reply src/main/containers/TestPanel.jsx:14]: Ah good catch—yes, it was commented out for testing while we worked out the ID issues. Should just need a little refactoring to increment the tree count like we did in the source panel

File: src/main/containers/SourcePanel.jsx (hunk 2 above)
Comment (conor-jh) [reply src/main/containers/SourcePanel.jsx]: I’m confused by this—I’ve experienced errors before when using the extension… whereas when I omit it, things seem to work and I’m not _exactly_ sure why…

Comment (conor-jh) [reply src/main/containers/SourcePanel.jsx]: Scratch that, must’ve been a fluke. Will add in to all imports.

**Review by Donaldblanc — APPROVED** [review]
🥇  thanks for making those changes, looks great

**Pull request merged**